### PR TITLE
Include cluster in health appender request key

### LINKED ISSUE #6185: Multi-cluster graph - error when user lacks access to remote cluster
Given two clusters, `east` and `west`, with a `bookinfo` namespace in both and a user that has access to `bookinfo.east` but not `bookinfo.west`, Kiali sometimes shows the graph and sometimes shows an access error. There seems to be some access check that is performed but the exact cluster that it is performed against varies. The graph should only show namespaces to the user that they have access to but should not throw an error when that namespace exists on both clusters.

namespace_access_error

### PR DESCRIPTION
**Pull request by nrfox**
Key did not include cluster and when multiple namespaces exist across clusters the request was being overwritten.

Fixes #6185

### CODE REVIEW

**Comment by nrfox** [comment]
I should clarify that this doesn't actually fix the forbidden issue. That is caused by #6198. But it does fix the issue where sometimes you could see a graph for namespaces in a remote cluster that you didn't have access to.

**Comment by nrfox** [comment]
@leandroberetta when you get a chance would you mind taking a look at this?

**Review by leandroberetta — APPROVED** [review]
Looks good.

File: graph/telemetry/istio/appender/health.go [hunk 1]
```diff
@@ -182,7 +183,7 @@ func (a *HealthAppender) attachHealth(trafficMap graph.TrafficMap, globalInfo *g
 
 		var req healthRequest
 		var ok bool
-		if req, ok = healthReqs[n.Namespace]; !ok {
+		if req, ok = healthReqs[n.Namespace+n.Cluster]; !ok {
```
Comment (jshaughn) [inline graph/telemetry/istio/appender/health.go:186]: Minor nit, across the graph appender code, when needing a key like this, the format used has been `cluster:namespace`.  In general, it's safer to use a delimiter when combining strings, and conventionally the cluster has been left of the namespace in the graph-related keys.

**Pull request merged**